src/wp-includes/rest-api/class-wp-rest-server.php

```diff
@@ -565,10 +565,22 @@ public function serve_request( $path = null ) {
 	 * }
 	 */
 	public function response_to_data( $response, $embed ) {
-		$data  = $response->get_data();
-		$links = self::get_compact_response_links( $response );
+		$data              = $response->get_data();
+		$links             = self::get_compact_response_links( $response );
+		$is_link_requested = false;
 
-		if ( ! empty( $links ) ) {
+		if ( ! empty( $_GET['_fields'] ) ) {
```

Comment:
`$_GET` is not going to be reliable for checking `_fields` because REST API requests can be dispatched within PHP, not just with the browser.

Also, I think relying on `$_GET` here would mean that all responses from the server would use the same set of fields, ignoring the `_fields` passed to individual requests, which I don't think is the desired outcome.

I think we'll want to use the fields that are in the request object, not globals.

Reply:
Thanks for the updates! I'll try to review the PR again this weekend.